geode/src/main/java/org/apache/calcite/adapter/geode/rel/GeodeFilter.java

```diff
@@ -233,6 +238,90 @@ private String translateOp2(String op, String name, RexLiteral right) {
       }
       return name + " " + op + " " + valueString;
     }
+
+    /**
+     *  Get the field name for the left node to use for IN SET query
+     */
+    private String getLeftNodeFieldName(RexNode left) {
+      switch (left.getKind()) {
+      case INPUT_REF:
+        final RexInputRef left1 = (RexInputRef) left;
+        return fieldNames.get(left1.getIndex());
+      case CAST:
+        // FIXME This will not work in all cases (for example, we ignore string encoding)
+        return getLeftNodeFieldName(((RexCall) left).operands.get(0));
+      case OTHER_FUNCTION:
+        return left.accept(new GeodeRules.RexToGeodeTranslator(this.fieldNames));
+      default:
+        return null;
+      }
+    }
+
+    /**
+     *  Check if we can use IN SET Query clause to improve query performance
+     */
+    private boolean useInSetQueryClause(List<RexNode> disjunctions) {
+      Set<String> leftFieldNameSet = new HashSet<>();
+
+      return disjunctions.stream().allMatch(node -> {
+        // IN SET query can only be used for EQUALS
+        if (node.getKind() != SqlKind.EQUALS) {
+          return false;
+        }
+
+        RexCall call = (RexCall) node;
+        final RexNode left = call.operands.get(0);
+        final RexNode right = call.operands.get(1);
+
+        // The right node should always be literal
+        if (right.getKind() != SqlKind.LITERAL) {
+          return false;
+        }
+
+        String name = getLeftNodeFieldName(left);
+        if (name == null) {
+          return false;
+        }
+
+        leftFieldNameSet.add(name);
+
+        // Ensure that left node field name is same for Nodes in disjunctions
+        if (leftFieldNameSet.size() != 1) {
+          return false;
+        }
+
+        return true;
+      });
+    }
+
+    /**
+     * Creates OQL IN SET predicate string
+     */
+    private String translateInSet(List<RexNode> disjunctions) {
+      RexNode firstNode = disjunctions.get(0);
+      RexCall firstCall = (RexCall) firstNode;
+
+      final RexNode left = firstCall.operands.get(0);
+      String name = getLeftNodeFieldName(left);
+
+      Set<String> rightLiteralValueList = new HashSet<>();
+
+      disjunctions.forEach(node -> {
+        RexCall call = (RexCall) node;
+        RexLiteral rightLiteral = (RexLiteral) call.operands.get(1);
+
+        rightLiteralValueList.add(quoteCharLiteral(rightLiteral));
+      });
+
+      final StringBuilder stringBuilder = new StringBuilder();
+
+      stringBuilder.append(name);
+      stringBuilder.append(" IN SET(");
+      stringBuilder.append(Util.toString(rightLiteralValueList, "", ", ", ""));
```

Comment:
[String.join()](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#join-java.lang.CharSequence-java.lang.Iterable-) should be sufficient here. `Util` class is pre java 8.